_includes/v22.2/start-in-docker/mac-linux-steps-single-node.md

```diff
@@ -0,0 +1,128 @@
+    - If the `/cockroach/cockroach-data` within the container is empty, the specified database, user, and password are created automatically. Instead of specifying each value directly by using the `-e` or `--env` flag, you can store them in a file on the Docker host. Use one key-value pair per line and set the `--env-file` flag to the file's path.
+    - Bind-mounts the `~/init-scripts` directory on the Docker host onto the `/docker-entrypoint-initdb.d` directory within the container. Initialization scripts stored in this directory are run after CockroachDB starts and the default database, user, and password are initialized.
+    - Accepts database client connections on IP address 172.18.0.3 (which resolves to the hostname `roach-single` within the Docker container) on port 26257.
+    - Accepts connections to the DB Console on IP address 172.18.0.3 (which resolves to the hostname `roach-single` within the Docker container) on port 8080.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    docker run -d \
+              --env COCKROACH_DATABASE={DATABASE_NAME} \
+              --env COCKROACH_USER={USER_NAME} \
+              --env COCKROACH_PASSWORD={PASSWORD} \
+              --name=roach-single \
+              --hostname=roach-single \
+              --ip=172.18.0.3 \
+              --advertise-addr=172.18.0.3 \
+              -p 26257:26257 -p 8080:8080 \
+              -v "roach-single:/cockroach/cockroach-data" \
+              -v "~/init-scripts:/docker-entrypoint-initdb.d" \
+              cockroachdb/cockroach:latest start-single-node
+    ~~~
+
+    By default, a `certs` directory is created and CockroachDB starts in secure mode. To prevent this, add the `--insecure` flag after the `start-single-node` sub-command.
+
+    {{site.data.alerts.callout_info}}
+    The `COCKROACH_DATABASE`, `COCKROACH_USER`, and `COCKROACH_PASSWORD` environment variables and the contents of the `/docker-entrypoint-initdb.d` directory are ignored if you use `cockroach start` rather than `cockroach start-single-node`. They are also ignored if the `/cockroach/cockroach-data` directory within the container is not empty.
+    {{site.data.alerts.end}}
+
+1. Perform a one-time initialization of the cluster:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    docker exec -it roach-single ./cockroach init --insecure
+    ~~~
+
+    You'll see the following message:
+
+    ~~~
+    Cluster successfully initialized
+    ~~~
+
+    At this point, the cluster node also prints helpful [startup details](cockroach-start.html#standard-output) to its log. For example, the following command retrieves node 1's startup details:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    docker exec -it roach-single grep 'node starting' cockroach-data/logs/cockroach.log -A 11
+    ~~~
+
+    The output will look something like this:
+
+    ~~~
```

Comment:
I think this output is for an insecure node, but not for a secure node as in the `Start the cluster node` example. 

The output for a secure node (for v22.2.0-beta.1) seems to be 
```
CockroachDB node starting at 2022-09-28 14:35:27.495508396 +0000 UTC m=+1.370085757 (took 0.5s)
build:               CCL v22.2.0-beta.1 @ 2022/09/26 18:49:07 (go1.19.1)
webui:               https://127.0.0.1:8080
sql:                 postgresql://root@127.0.0.1:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt
sql (JDBC):          jdbc:postgresql://127.0.0.1:26257/defaultdb?sslcert=certs%2Fclient.root.crt&sslkey=certs%2Fclient.root.key&sslmode=verify-full&sslrootcert=certs%2Fca.crt&user=root
RPC client flags:    /cockroach/cockroach <client cmd> --host=127.0.0.1:26257 --certs-dir=certs
logs:                /cockroach/cockroach-data/logs
temp dir:            /cockroach/cockroach-data/cockroach-temp2611102055
external I/O path:   /cockroach/cockroach-data/extern
store[0]:            path=/cockroach/cockroach-data
storage engine:      pebble
clusterID:           60f29a4e-1c87-4b0c-805d-eb73460766b1
status:              initialized new cluster
nodeID:              1
```

Reply:
This output also makes me question the need to create a Docker network, since you are reaching the cluster at `127.0.0.1:26257`. Are these in conflict?

Reply:
I updated the command like this and I added a bit of text explaining that Docker adds a DNS rule (I think the DNS rule is based on the `--name` flag actually, and that the `--hostname` flag should be used to change the hostname within the container if it should be different from `--name`. Does CRDB require setting both?)